src/material.js

```diff
@@ -158,12 +162,26 @@ export class Material {
         });
     }
 
+    set Basis(config) {
+        const newConfig = {
+            elements: config.elements,
+            coordinates: config.coordinates,
+            units: config.units,
+            cell: config.cell,
+            isEmpty: false
+        };
+        return new ConstrainedBasis({
+                ...newConfig
+            }
+        );
+    }
```

Comment:
![](https://media2.giphy.com/media/11ahZZugJHrdLO/giphy.gif)

A setter that returns a Class instance???

Reply:
Let's remove this pls